FIRO_TSEnsembles/src/main/java/hec/stats/MultiStatComputable.java

```diff
@@ -25,21 +35,30 @@ public float[] MultiCompute(float[] values) {
         }
 
         for (int i = 0; i < size; i++) {
-
-            switch (statSelection[i]){
-                case MIN:
-                    results[i] = is.getMin();
-                    break;
-                case MEAN:
-                    results[i] = is.getMean();
-                    break;
-                case MAX:
-                    results[i] = is.getMax();
-                    break;
-                default:
-                    throw new ArithmeticException("stat type not  yet supported in MultiStatComputable.");
-            }
-
+            results[i] = selectedStatCompute(i, is);
+        }
+        return results;
+    }
+    public float selectedStatCompute(int selectedStat, InlineStats is) {
```

Comment:
nice refactor!  consider making selectedStatCompute() private if it is only used within this class.